code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -584,9 +578,11 @@
 
 	if(X == living_xeno_ruler)
 		on_ruler_death(X)
-
-	return TRUE
-
+	var/datum/xeno_caste/caste = X?.xeno_caste
+	if(X in xenos_by_typepath[X.caste_base_type] && caste.death_evolution_delay > 0)
```

Comment:
You're only setting up a death timer if the caste's evo delay is <=0, isn't that backwards? Also not sure what the in xenos_by_typepath check is for.

Reply:
The first part was originally in the queen's code I think, wasn't sure why it was there, so I just moved it here. And yeah, you're correct, I did that wrong